iceoryx_examples/ara_com/minimal_proxy.hpp

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "topic.hpp"
+
+#include "ara/com/event_subscriber.hpp"
+#include "ara/com/field_subscriber.hpp"
+#include "ara/com/method_client.hpp"
+#include "ara/runtime.hpp"
+#include "ara/types.hpp"
+
+class MinimalProxy
+{
+  public:
+    /// @todo make c'tor of cxx::string constexpr'able
+    static constexpr char m_serviceIdentifier[] = "MinimalSkeleton";
+
+    MinimalProxy(ara::com::ProxyHandleType& handle)
+        : m_instanceIdentifier(handle.GetInstanceId())
+    {
+        if (handle.GetServiceId() != ara::com::ServiceIdentifier{iox::cxx::TruncateToCapacity, m_serviceIdentifier})
+        {
+            std::cout << "Handle does not match MinimalProxy class. Can't construct MinimalProxy, terminating!"
+                      << std::endl;
+            std::terminate();
+        }
+    }
+    MinimalProxy(const MinimalProxy&) = delete;
+    MinimalProxy& operator=(const MinimalProxy&) = delete;
+
+    static ara::com::FindServiceHandle StartFindService(ara::com::FindServiceHandler<ara::com::ProxyHandleType> handler,
+                                                        ara::com::InstanceIdentifier& instanceIdentifier) noexcept
+    {
+        ara::com::ServiceIdentifier serviceIdentifier{iox::cxx::TruncateToCapacity, m_serviceIdentifier};
+        return ara::Runtime::GetInstance().StartFindService(handler, serviceIdentifier, instanceIdentifier);
+    }
+
+    static void StopFindService(ara::com::FindServiceHandle handle) noexcept
+    {
+        ara::Runtime::GetInstance().StopFindService(handle);
+    }
+
+    static ara::com::ServiceHandleContainer<ara::com::ProxyHandleType>
+    FindService(ara::com::InstanceIdentifier& instanceIdentifier) noexcept
+    {
+        ara::com::ServiceIdentifier serviceIdentifier{iox::cxx::TruncateToCapacity, m_serviceIdentifier};
+        return ara::Runtime::GetInstance().FindService(serviceIdentifier, instanceIdentifier);
```

Comment:
I think there's a missunderstanding here. What you describe is exactly what's implemented atm. Whether a AUTOSAR service maps to several iceoryx services should be implemented with a database query e.g. via `Runtime::verifyThatServiceIsComplete()`. Counting is ok for a brief `ara::com` example IMHO.

Reply:
@mossmaurice The point is, it is not really up to the runtime to know whether the service is complete (and yet you do the counting in the runtime). The service class should be the one who knows this.

Reply:
@MatthiasKillat Sure, that's why the idea is that `Runtime::verifyThatServiceIsComplete()` calls some external method e.g. in a database object or whoever knows when a service is complete. Isn't that clear from the comment?

Reply:
Ideally it would not need a data base. This should all be resolvable on service level once the all the names (service, instance, events etc.) are known at runtime.